src/main/java/org/manifold/compiler/front/ConstantValueVertex.java

```diff
@@ -0,0 +1,56 @@
+package org.manifold.compiler.front;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+
+import org.manifold.compiler.TypeValue;
+import org.manifold.compiler.Value;
+
+public class ConstantValueVertex extends ExpressionVertex {
```

Comment:
I don't exactly understand the purpose of converting the `Value` and `Expression` objects into `ValueVertex` and `ExpressionEdge` objects. What do the `Vertex` and `Edge` classes do that the `Value` and `Expression` classes do not? I think I've complained about this before, sorry! I think it adds a huge amount of conceptual complexity to this project.